python/grass/gunittest/runner.py

```diff
@@ -443,55 +388,26 @@ def setTimes(self, start_time, end_time, time_taken):
                     raise
 
 
-class GrassTestRunner:
+class GrassTestRunner(unittest.TextTestRunner):
+    resultclass = TextTestResult
+
     def __init__(
         self,
-        stream=None,
-        descriptions=True,
-        verbosity=1,
-        failfast=False,
-        buffer=False,
-        result=None,
-        resultclass=None,
-        warnings=None,
         *,
-        tb_locals=False,
+        result=None,
         **kwargs,
-    ):
-        if stream is None:
-            stream = sys.stderr
-        self.stream = _WritelnDecorator(stream)
-        self.descriptions = descriptions
-        self.verbosity = verbosity
-        self.failfast = failfast
-        self.buffer = buffer
-        self.tb_locals = tb_locals
-        self.warnings = warnings
-        if resultclass is not None:
-            self.resultclass = resultclass
+    ) -> None:
+        super().__init__(**kwargs)
 
         self._result = result
 
     def run(self, test):
         """Run the given test case or test suite."""
-        result = self._result
-        unittest.registerResult(result)
-        result.failfast = self.failfast
-        result.buffer = self.buffer
-        startTime = time.time()
-        startTestRun = getattr(result, "startTestRun", None)
-        if startTestRun is not None:
-            startTestRun()
-        try:
-            test(result)
-        finally:
-            stopTime = time.time()
-            timeTaken = stopTime - startTime
-            setTimes = getattr(result, "setTimes", None)
-            if setTimes is not None:
-                setTimes(startTime, stopTime, timeTaken)
-            stopTestRun = getattr(result, "stopTestRun", None)
-            if stopTestRun is not None:
-                stopTestRun()
-
+        startTime = time.perf_counter()
+        result = super().run(test)
```

Comment:
The previous code was trying to counter any exceptions. Is the unittest run doing something like that? I suppose it does.